dartagnan/src/main/java/com/dat3m/dartagnan/encoding/ExpressionEncoder.java

```diff
@@ -110,106 +109,92 @@ public Formula visitIntLiteral(IntLiteral intLiteral) {
     public Formula visitIntBinaryExpression(IntBinaryExpr iBin) {
         Formula lhs = encode(iBin.getLeft());
         Formula rhs = encode(iBin.getRight());
+        final int bitWidth = iBin.getType().isMathematical() ? 64 : iBin.getType().getBitWidth();
```

Comment:
Why the hardcoded constant? Don't we have a method somewhere to get the arch type width?

Reply:
The constant is somewhat arbitrary. The former code used `32`, but theoretically any constant is wrong I believe.
Since we never generate mathematical integers in any parser anymore (we used to with Boogie), the code is actually never executed.
We might even get rid of the mathematical integer type altogether...

Reply:
Are you finally getting rid of this?

Reply:
Yes. I just didn't have time yesterday